path_roles.go

```diff
@@ -167,30 +188,43 @@ func (b *tfBackend) pathRolesWrite(ctx context.Context, req *logical.Request, d
 		return logical.ErrorResponse("ttl cannot be greater than max_ttl"), nil
 	}
 
+	// if we're creating a role to manage a Team or Organization, we need to
+	// create the token now. User tokens will be created when credentials are
+	// read.
+	if roleEntry.Organization != "" || roleEntry.TeamID != "" {
+		token, err := b.createToken(ctx, req.Storage, roleEntry)
+		if err != nil {
+			return nil, err
+		}
+
+		roleEntry.Token = token.Token
+		roleEntry.TokenID = token.ID
+	}
+
 	if err := setTerraformRole(ctx, req.Storage, name, roleEntry); err != nil {
 		return nil, err
 	}
 
-	return &resp, nil
+	return nil, nil
 }
 
 func (b *tfBackend) pathRolesDelete(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
-	err := req.Storage.Delete(ctx, "roles/"+d.Get("name").(string))
+	err := req.Storage.Delete(ctx, "role/"+d.Get("name").(string))
 	if err != nil {
-		return nil, errwrap.Wrapf("error deleting terraform role: {{err}}", err)
+		return nil, fmt.Errorf("error deleting terraform role: %w", err)
 	}
 
 	return nil, nil
 }
 
 func setTerraformRole(ctx context.Context, s logical.Storage, name string, roleEntry *terraformRoleEntry) error {
-	entry, err := logical.StorageEntryJSON("roles/"+name, roleEntry)
+	entry, err := logical.StorageEntryJSON("role/"+name, roleEntry)
 	if err != nil {
 		return err
 	}
 
 	if entry == nil {
-		return errwrap.Wrapf("nil result writing to storage", nil)
+		return fmt.Errorf("nil result writing to storage")
```

Comment:
This is not actually performing the write (the `s.Put` call below is), so a more accurate error message would be:

```suggestion
		return fmt.Errorf("failed to create storage entry for role")
```